src/besseli.jl

```diff
@@ -130,6 +130,238 @@ function besseli1x(x::T) where T <: Union{Float32, Float64}
     return z
 end
 
+#####
+#####  Implementation for complex arguments
+#####
+
+function besseli0(z::ComplexF64)
+    check = false
+    if real(z) < 0.0
+        z = -z
+    end
+    if imag(z) < 0.0
+        z = conj(z)
+        check = true
+    end
+
+    if abs(z) > 17.5
+        zinv = 1 / z
+        e = exp(z)
+        s = sqrt(2 * z * π)
+        p = evalpoly(zinv*zinv, (1.0, 0.0703125, 0.112152099609375, 0.5725014209747314, 6.074042001273483, 110.01714026924674, 3038.090510922384, 118838.42625678325, 6.252951493434797e6, 4.259392165047669e8, 3.646840080706556e10, 3.8335346613939443e12, 4.8540146868529006e14, 7.286857349377656e16))
+        p2 = zinv*evalpoly(zinv*zinv, (0.125, 0.0732421875, 0.22710800170898438, 1.7277275025844574, 24.380529699556064, 551.3358961220206, 18257.755474293175, 832859.3040162893, 5.0069589531988926e7, 3.8362551802304335e9, 3.6490108188498334e11, 4.218971570284097e13, 5.827244631566907e15, 9.47628809926011e17))
+        r = e / s * (p + p2) + im * (p - p2) / (e * s)
```

Comment:
you might be able to squeeze a little more accuracy by removing the `zinv` from `p2` and then combining `p` and `p2` with a `muladd`.